lib/gradient/elixir_checker.ex

```diff
@@ -0,0 +1,88 @@
+defmodule Gradient.ElixirChecker do
+  @moduledoc ~s"""
+  Provide checks specific to Elixir that complement type checking delivered by Gradient.
+
+  Options:
+  - {`ex_check`, boolean()}: whether to use checks specific only to Elixir.
+  """
+
+  @spec check([:erl_parse.abstract_form()], keyword()) :: [{:file.filename(), any()}]
+  def check(forms, opts) do
+    if Keyword.get(opts, :ex_check, true) do
+      check_spec(forms)
+    else
+      []
+    end
+  end
+
+  @doc ~s"""
+  Check if all specs are exactly before the function that they specify
+  and if there is only one spec per function clause.
+
+  Correct specs location:
+    ```
+    @spec convert(integer()) :: float()
+    def convert(int) when is_integer(int), do: int / 1
```

Comment:
Just inserting a newline for clarity.

```suggestion
    def convert(int) when is_integer(int), do: int / 1
    
```